staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -270,6 +272,10 @@ func transformDecodeError(typer runtime.ObjectTyper, baseErr error, into runtime
 // setSelfLink sets the self link of an object (or the child items in a list) to the base URL of the request
 // plus the path and query generated by the provided linkFunc
 func setSelfLink(obj runtime.Object, requestInfo *request.RequestInfo, namer ScopeNamer) error {
+	if utilfeature.DefaultFeatureGate.Enabled(features.RemoveSelfLink) {
```

Comment:
are setSelfLink and setObjectSelfLink called for every item in a list response? checking a feature gate here seems like it might be really expensive